docs/proposals/quic-design.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Quic Design
+status: Alpha
+authors:
+    - "@Qianchenjie"
+approvers:
+creation-date: 2019-05-01
+last-updated: 2019-05-01
+---
+
+# Enhanced Cloud/Edge Communication
+
+## Abstract
+In order to enhance cloud and edge communication efficiency, we introduced the QUIC protocol for cloudhub and edgehub.
+I propose that cloudHub should support both websocket and QUIC protocol access at the same time. And the edgehub can choose one of the protocols to access to the cloudhub.
+QUIC is a new transport which reduces latency compared to that of TCP. On the surface, QUIC is very similar to TCP+TLS+HTTP/2 implemented on UDP. Because TCP is implemented in operating system kernels, and middlebox firmware,making significant changes to TCP is next to impossible.However,since QUIC is built on top of UDP,it suffers from no such limitations.
+Key features of QUIC:
+
+1. Dramatically reduced connection establishment time
+2. Improved congestion control
+3. Multiplexing without head of line blocking
+4. Forward error correction
+5. Connection migiation
+![此处输入图片的描述][1]
+![此处输入图片的描述][2]
+
+## Motivation
+For the edge side, the node network signal instability, TCP + TLS overhead is very large, QUIC'S 0-RTT can greatly increase access speed. 
+Due to the characteristics of multiplexing, business processing capacity can be greatly improved.
+
+## Use cases
+<img src="../images/proposals/quic-design.png">
+
+### Start the websocket server
+1. User edit controller.yaml, modify cloudhub.protocol_websocket: true, and port: 1000.
+2. Run the edgecontroller, and start the websocket server.
+3. Waiting for client access through websocket protocol.
+
+### Start the quic server
+1. User edit controller.yaml, modify cloudhub.protocol_quic: true, and quic_port: 1001
+2. Run the edgecontroller, and start the quic server.
+3. Waiting for client access through quic protocol.
+
+### Start the websocket and quic server
+1. User edit controller.yaml, modify cloudhub.protocol_quic: true, and quic_port: 1001， cloudhub.protocol_websocket: true, and port: 1000
+2. Run the edgecontroller, and start the quic and websocket server.
+3. Waiting for client access through quic and websocket protocol.
+
+### edgehub connect to cloudhub through websocket protocol
+1. User edit edge.yaml, modify edgehub.controller.protocol: websocket.
+2. Run the edgecore, and start to connect to cloudhub through websocket protocol.
+
+### edgehub connect to cloudhub through quic 
+1. User edit edge.yaml, modify edgehub.controller.protocol: quic.
+
+        quic:
+            url: 127.0.0.1:10001
```

Comment:
10001 or 1001?

Reply:
@m1093782566 , we use 10000 for websocket,

Reply:
I mean we should keep consistent across the examples.

Reply:
we should support quic and websocket at the same time. so 10000 for websocket, 10001for quic.